### DISCUSSION TITLE
STM32CubeProgrammer: Warning: File corrupted. Two or more segments defines the same memory zone

### DISCUSSION THREAD

**Discussion by RobbertProost** [modm-io/modm#922] [Q&A]
When I am building an example project for the STM32 Nucleo-L476RG (for example 'blinky') and I load the resulting .elf file with the STM32CubeProgrammer it gives me a warning: File corrupted. Two or more segments defines the same memory zone

It however still allows me to flash the firmware and when running it, it seems to run fine.
I am I doing something wrong, or what could be the reason for this warning?

**Comment by salkinium** [comment]
Oh, interesting! I didn't know about the STM32CubeProgrammer until now, unfortunately I cannot get it to work on my machine.

We use a more complex linkerscript (see the generated `modm/link/linkerscript.ld` for your target) than what CubeMX usually generates.
It could be a bug in our linkerscript that uses a lot of different sections at different locations and load addresses, or it could be a bug in the STM32CubeProgrammer's interpretation of the ELF sections.

I would recommend using OpenOCD instead, which also gives you all the other features such as debugging, RTT and ITM logging.

**Reply by chris-durand** [reply]
It's very likely our fault. We are having the same issues with openocd 0.12.0 (#985).
